### PR TITLE
fix: prioritize combobox value over label

### PR DESCRIPTION
**Pull request by anubra266**
## 📝 Description

Use option values for component logic

## ⛳️ Current behavior (updates)

Currently using option label

### CODE REVIEW

File: packages/machines/combobox/src/combobox.utils.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { dom } from "./combobox.dom"
+import type { MachineContext as Ctx } from "./combobox.types"
+
+export const utils = {
+  labelFromValue: (ctx: Ctx, value: string) => {
+    const el = dom.getMatchingOptionEl(ctx, value)
+    const data = dom.getOptionData(el)
+    return data.label
+  },
+}
```
Comment (segunadebayo) [inline packages/machines/combobox/src/combobox.utils.ts:10]: Please move this to the dom file and rename it to `getValueLabel`

File: .changeset/sweet-taxis-agree.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@zag-js/combobox": patch
+---
+
+give value higher priority over label.
```
Comment (segunadebayo) [inline .changeset/sweet-taxis-agree.md:5]: Can you re-word this to explain the problem it solves?

File: packages/machines/combobox/src/combobox.connect.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ export function connect<T extends PropTypes = ReactPropTypes>(state: State, send
     activeOption: state.context.activeOptionData,
     selectedValue: state.context.selectedValue,
     setValue(value: string) {
-      send({ type: "SET_VALUE", value })
+      send({ type: "SET_VALUE", value, isOptionValue: true })
```
Comment (segunadebayo) [inline packages/machines/combobox/src/combobox.connect.ts:37]: Any reason for the `isOptionValue: true` here?

**Review by segunadebayo — CHANGES_REQUESTED** [review]
Thanks for this. Can you also run the tests to ensure it's all working?

File: packages/machines/combobox/src/combobox.connect.ts (hunk 3 above)
Comment (anubra266) [reply packages/machines/combobox/src/combobox.connect.ts:37]: @segunadebayo the function that handles `setValue` also handles the input `onChange` so we need to differentiate when the event value is the option `value` from when it's the `label`, so when it's value, we can get the label first before changing the input value.

**Comment by anubra266** [comment]
> Thanks for this. Can you also run the tests to ensure it's all working?
@segunadebayo Tests are working fine
CleanShot 2022-06-19 at 18 21 52@2x

**Pull request merged**